# Generated by Django 4.2 on 2025-04-04 19:26

import clinica.models
import django.core.validators
from django.db import migrations, models
import django.db.models.functions.datetime


class Migration(migrations.Migration):

    dependencies = [
        ('clinica', '0001_initial'),
    ]

    operations = [
        migrations.AlterField(
            model_name='medico',
            name='crm',
            field=models.CharField(max_length=8, unique=True, validators=[clinica.models.Medico.formatador]),
        ),
        migrations.AlterField(
            model_name='medico',
            name='email',
            field=models.EmailField(blank=True, max_length=254),
        ),
        migrations.AlterField(
            model_name='medico',
            name='especialidade',
            field=models.CharField(choices=[('CAR', 'Cardiologista'), ('PSI', 'Psiquiatra')], max_length=30),
        ),
        migrations.AlterField(
            model_name='medico',
            name='nome',
            field=models.CharField(max_length=50, validators=[django.core.validators.MinLengthValidator(5)]),
        ),
        migrations.AddConstraint(
            model_name='consulta',
            constraint=models.CheckConstraint(check=models.Q(('data__gte', django.db.models.functions.datetime.Now())), name='consulta_nao_pode_ser_em_datas_passadas'),
        ),
    ]
